### PR TITLE
Initial attempt at integration tests

### PR DESCRIPTION
**Pull request by will-in-wi**
Based on https://github.com/DripEmail/magento-m1-extension/pull/8

### CODE REVIEW

File: devtools_m2/package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "drip-magento2-tests",
+  "version": "1.0.0",
+  "main": "index.js",
+  "license": "MIT",
+  "dependencies": {
```
Comment (jachin) [inline devtools_m2/package.json]: Are these dependencies are are they "dev dependencies".

https://medium.com/@dylanavery720/npmmmm-1-dev-dependencies-dependencies-8931c2583b0c

File: devtools_m2/setup.sh [hunk 2]
```diff
@@ -31,7 +31,30 @@ MAGE_MODE=developer ./bin/magento setup:install \
 SCRIPT
 )
 
-docker-compose exec -T -u www-data web /bin/bash -c "$magento_setup_script"
+./docker_compose.sh exec -T -u www-data web /bin/bash -c "$magento_setup_script"
 
-# Backup for reset.
-docker-compose exec -e MYSQL_PWD=magento db mysqldump -u magento magento > db_data/dump.sql
+# For multi-store.
+./docker_compose.sh exec -T -u www-data web patch -p1 /var/www/html/magento/index.php <<'SCRIPT'
```
Comment (jachin) [inline devtools_m2/setup.sh]: Doing a patch seems brittle... but maybe it's good for it to break quickly.

File: devtools_m2/package.json (hunk 1 above)
Comment (will-in-wi) [reply devtools_m2/package.json]: I was thinking that the entire application exists to run tests. So there are no "dev" dependencies, really. But I'm not sure whether there is some other difference.

File: devtools_m2/setup.sh (hunk 2 above)
Comment (will-in-wi) [reply devtools_m2/setup.sh]: That was my thought. This change is the official way to enable multi-site (although I might be able to do something with environment variables in some cases). The patch will break immediately if they change index.php.

**Pull request merged**